numpy/core/tests/test_umath.py

```diff
@@ -686,6 +686,49 @@ def test_float_remainder_corner_cases(self):
                 assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, rem))
 
 
+class TestDivisionOverflows:
+    @pytest.mark.parametrize("dividend_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("divisor_dtype",
+            np.sctypes['int'])
+    @pytest.mark.parametrize("operation",
+            [np.remainder, np.fmod, np.floor_divide])
+    def test_scalars(self, dividend_dtype, divisor_dtype, operation):
+        with np.errstate(divide='raise'):
+            # If dividend is a larger type than the divisor (`else` case),
+            # then, result will be a larger type than dividend and will not
+            # result in an overflow.
+            if np.dtype(dividend_dtype).itemsize >= np.dtype(
```

Comment:
TBH, I expected arrays to satisfy this `if`, but it did not. Not sure if it's a bug or it's intended. Leaving this here to confirm.

Reply:
I think you are running into value based-casting?  Make sure the devisor is not 0-D (or scalar).

I.e. the simplest solution, will be to just write divisor_dytpe([-1])` even if that is a bit funny, or maybe move it out, and write `divisor = np.array(-1, dtype=divisor_dtype)`.

About the floating-point warning/error.  I am OK with this being given, but if it is easy to avoid. `fmod` and `remainder` do not actually overflow.  `0` is the correct result without any warning.
(But I am willing to ignore it in case the warning is triggered through `divmod` or so.)

Reply:
Thanks for the review Sebastian!

> `divisor_dytpe([-1])`

Ah ok, thanks for this, will try it out.

> fmod and remainder do not actually overflow

I referred https://stackoverflow.com/a/19285537/5671364, where we see the modulo operator leading to overflow and this does seem to be the case in a few systems. I have captured a few more details here: https://github.com/numpy/numpy/issues/21506

Reply:
Well, methematically it seems clear that `x % -1 == 0` since `x // -1 == -x` and `x - (-x * -1) == 0`.

To me, the minimum integer seems special for the division (due to not being representable), but not for the remainder.
Instruction/hardware wise, that may be different.  But unless we lose a lot of performance or it is very inconvenient, I would tend to not signal overflow for the remainder.

Reply:
Ahh ok, got it. We will handle the cases but not raise an overflow error for remainder as the result is actually 0, unlike the division case. Will make that change, thanks for clarifying!

Reply:
There is a slight discrepancy between arrays and scalars in divmod and floor_divide for raising errors. Will take a look at it soon. Tests are in place.